### ISSUE TITLE
Regression: FileRepo.py doesn't follow symlinks

### ISSUE THREAD

**Issue by szuta** [munki/munki#791]
See #473. FileRepo.py is not set to follow symlinks.

**Comment by szuta** [comment]
The `itemlist` function of `FileRepo.py` uses  `os.walk`, which does not follow symlinks. So if our pkgsinfo directory contains symlinks to other directories, they won't be picked up during a `makecatalogs` run.  A similar issue occured in #473 and was fixed in f4769888162fabe1c14045fc56f6d7beb52bbdcd.

Here's a fix that works on our setup, but I'm not sure if it suffices to recurse only through `pkgsinfo` and `pkgs`:

```
--- a/code/client/munkilib/munkirepo/FileRepo.py
+++ b/code/client/munkilib/munkirepo/FileRepo.py
@@ -189,8 +189,9 @@ class FileRepo(Repo):
         kind = unicodeize(kind)
         search_dir = os.path.join(self.root, kind)
         file_list = []
+        followlinks = (kind in ['pkgsinfo', 'pkgs'])
         try:
-            for (dirpath, dirnames, filenames) in os.walk(search_dir):
+            for (dirpath, dirnames, filenames) in os.walk(search_dir, followlinks):
                 # don't recurse into directories that start with a period.
                 dirnames[:] = [name
                                for name in dirnames if not name.startswith('.')]
```

**Comment by gregneagle** [comment]
We'll just follow symlinks in all the repo directories and leave it to the admin to not do something crazy/stupid.  See https://github.com/munki/munki/commit/744840d85a784ee2b653cf6670123682ac920065
